### ISSUE TITLE
Bcrypt causing CPU usage issue and consuming alot of time.

### ISSUE THREAD

**Issue by sharmankita** [kelektiv/node.bcrypt.js#922]
Hi, I am using bcrpyt.js to generate password in our system.
we have functionality of employee upload in this while inserting employees i am creating password for them as well.

so this process is working in loop.

but it taking a lot of time and CPU. like for 10000 employee my code took around 19 mins. attaching the sample code screenshot.

```
const crypto = require('crypto');
const bcrypt = require('bcryptjs');
let a = [];
console.time("dfsdf")
for(let i = 0 ; i < 10000 ; i++){
     a.push(createPassword())
}
console.timeEnd("dfsdf")


function createPassword () {
  let str = createRandomString(10);
  const saltRounds = 10;
  var enc_password = bcrypt.hashSync(str, saltRounds);
  return enc_password;
}


function createRandomString (len) {
  const buffer = crypto.randomBytes(len);
  var text = buffer.toString('hex');
  return text;
}
```

![Screenshot 2022-03-14 at 6 28 26 PM](https://user-images.githubusercontent.com/28458055/158176888-e6fbb86c-c7bf-4a72-9218-3c91d2326539.png)

**Comment by RSLak1** [comment]
Although I am not a contributer to this project, I can give you the answer:
Bcrypt is a hashing function with a so called "work factor" or "delay factor". It is intentionally "slow" (CPU- and memoryintensive). The slowness is what makes it secure as a hashing function for passwords.
As per [README.md](https://github.com/kelektiv/node.bcrypt.js), 10 salt rounds are roughly equivalent to about 10 hashes per second on a 2GHz core. 
10,000 passwords / 10 passwords per second = 1000 seconds = ~16.7min. 

For more information on why this is needed, I can suggest reading the [OWASP Password Storage Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html), especially the sections on [How Attackers Crack Password Hashes](https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html#how-attackers-crack-password-hashes) and on [Work Factors](https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html#work-factors).

TLDR: to me this looks like working as intended.

**Comment by recrsn** [comment]
As said by @RSLak1, It is well within speed expectations and bcrypt is CPU heavy

Also, you are using bcryptjs (the pure-JS bcrypt implementation) which is about 20-25% slower.

**Comment by sharmankita** [comment]
@RSLak1  Thanks for the information.
